js/bybit.js

```diff
@@ -1853,15 +2049,87 @@ module.exports = class bybit extends Exchange {
     }
 
     parseTrade (trade, market = undefined) {
+        const isSpotTrade = ('isBuyerMaker' in trade) || ('feeTokenId' in trade);
+        if (isSpotTrade) {
+            return this.parseSpotTrade (trade, market);
+        } else {
+            return this.parseContractTrade (trade, market);
+        }
+    }
+
+    parseSpotTrade (trade, market = undefined) {
         //
-        //  public spot
-        //
-        //    {
+        //   public:
+        //     {
         //        "price": "39548.68",
         //        "time": "1651748717850",
         //        "qty": "0.166872",
-        //        "isBuyerMaker": true
-        //    }
+        //        "isBuyerMaker": 0
+        //     }
+        //
+        //   private:
+        //     {
+        //         "orderPrice": "82.5",
+        //         "creatTime": "1666702226326",
+        //         "orderQty": "0.016",
+        //         "isBuyer": "0",
+        //         "isMaker": "0",
+        //         "symbol": "AAVEUSDT",
+        //         "id": "1274785101965716992",
+        //         "orderId": "1274784252359089664",
+        //         "tradeId": "2270000000031365639",
+        //         "execFee": "0",
+        //         "feeTokenId": "AAVE",
+        //         "matchOrderId": "1274785101865076224",
+        //         "makerRebate": "0",
+        //         "executionTime": "1666702226335"
+        //     }
+        //
+        const timestamp = this.safeIntegerN (trade, [ 'time', 'creatTime' ]);
+        let takerOrMaker = undefined;
+        let side = undefined;
+        const isBuyerMaker = this.safeInteger (trade, 'isBuyerMaker');
+        if (isBuyerMaker !== undefined) {
+            // if public response
+            takerOrMaker = 'taker'; // public trades are always taker
+            side = isBuyerMaker === 1 ? 'buy' : 'sell';
+        } else {
+            // if private response
+            const isBuyer = this.safeInteger (trade, 'isBuyer');
+            const isMaker = this.safeInteger (trade, 'isMaker');
+            takerOrMaker = isMaker === 1 ? 'maker' : 'taker';
+            side = isBuyer === 1 ? 'buy' : 'sell';
+        }
+        const marketId = this.safeString (trade, 'symbol');
+        market = this.safeMarket (marketId, market);
+        let fee = {};
+        const feeToken = this.safeString (trade, 'feeTokenId');
+        if (feeToken !== undefined) {
+            const feeCurrency = this.safeCurrencyCode (feeToken);
+            const feeCost = this.safeNumber (trade, 'execFee');
```

Comment:
this is crashing as well, I think we must parse it as a string here
![image](https://user-images.githubusercontent.com/43336371/200018549-041aa43b-0dc2-4dbe-be45-ce17b3a2217a.png)

Reply:
uh oh. when i ran private tests, this was not caught. fixed !